src/libra_py/workflows/nbra/lz.py

```diff
@@ -370,7 +487,9 @@ def run(H_vib, params):
                                 istate[Tr] = st_new                  
                         else:
                             istate[Tr] = st_new
-        
-    return res
 
+    if return_probabilities == True:

```

Comment:
do just 
"return res, P" and if you don't compute P - make it None - we shall not have the function returning different number of results depending on control parameters - that could break its usage patterns.